src/pyseekdb/client/client_base.py

```
@@ -333,6 +335,132 @@ def _truncate(val, length=20):
+        Get database object
+
+        Args:
+            name: database name
+            tenant: tenant name (for OceanBase)
+        """
+        effective_tenant = self._database_tenant(tenant)
+        logger.info(f"Getting database: {name}{self._database_context(effective_tenant)}")
+        sql = (
+            "SELECT SCHEMA_NAME, DEFAULT_CHARACTER_SET_NAME, DEFAULT_COLLATION_NAME "
+            "FROM information_schema.SCHEMATA "
+            f"WHERE SCHEMA_NAME = '{name}'"
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)